common/spec/warning_monkey_patch.rb

```diff
@@ -1,8 +1,9 @@
 # frozen_string_literal: true
 
 ALLOW_PATTERNS = [
-  # Ignore parser warnings for ruby 2.7 minor version mismatches
-  # TODO: Fix these by upgrading to ruby 2.7.3 (requires ubuntu upgrade)
+  # Ignore parser warnings for ruby 2.7 minor version mismatches.
+  # This is a recurring issue that occurs whenever the parser gets
+  # ahead of our installed ruby version.
   %r{parser/current is loading parser/ruby27},
   /2.7.\d-compliant syntax, but you are running 2.7.\d/,
```

Comment:
I am unclear why these first two lines are separate patterns, when the matching error line is a single line:
```
parser/current is loading parser/ruby27, which recognizes2.7.6-compliant syntax, but you are running 2.7.5.
```

Reply:
Just did a quick Google search for "compliant syntax, but you are running " and saw [an issue](https://github.com/whitequark/parser/issues/231) wherein someone had the message displayed as:

```ruby
warning: parser/current is loading parser/ruby22, which recognizes
warning: 2.2.3-compliant syntax, but you are running 2.2.4.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
```

I'd expect `ALLOW_PATTERNS.none? { |pattern| pattern =~ message }` to nicely handle that case and the [case you quoted](https://github.com/dependabot/dependabot-core/pull/5031/files#r856532794) 🙂

Reply:
Ah, that makes perfect sense, thank you @landongrindheim!